kyo-prelude/shared/src/main/scala/kyo/kernel/ArrowEffect.scala

```diff
@@ -163,21 +277,123 @@ object ArrowEffect:
+                            continue = handle4Loop(_, context)
+                        )
+                    case kyo: KyoSuspend[I3, O3, E3, Any, A, E1 & E2 & E3 & E4 & S & S2] @unchecked if effect3Tag =:= kyo.tag =>
+                        Safepoint.handle(kyo.input)(
+                            eval = handle3[Any](kyo.input, kyo(_, context)),
+                            suspend = handle4Loop(kyo, context),
+                            continue = handle4Loop(_, context)
+                        )
+                    case kyo: KyoSuspend[I4, O4, E4, Any, A, E1 & E2 & E3 & E4 & S & S2] @unchecked if effect4Tag =:= kyo.tag =>
+                        Safepoint.handle(kyo.input)(
+                            eval = handle4[Any](kyo.input, kyo(_, context)),
+                            suspend = handle4Loop(kyo, context),
+                            continue = handle4Loop(_, context)
+                        )
+                    case kyo: KyoSuspend[IX, OX, EX, Any, A, E1 & E2 & E3 & E4 & S & S2] @unchecked =>
+                        new KyoContinue[IX, OX, EX, Any, A, S & S2](kyo):
+                            def frame = _frame
+                            def apply(v: OX[Any], context: Context)(using Safepoint) =
+                                handle4Loop(kyo(v, context), context)
+                        end new
+                    case kyo =>
+                        kyo.asInstanceOf[A]
+                end match
+            end handle4Loop
+            handle4Loop(v, Context.empty)
+        end apply
     end handle
 
+    /** Handles an arrow effect with a stateful implementation. This combines function implementation with state management, allowing the
+      * handler to maintain and modify state as it processes operations. The state is threaded through the computation and can influence how
+      * operations are processed.
+      *
+      * @param effectTag
+      *   Identifies which arrow effect to handle
+      * @param state
+      *   The initial state value
+      * @param v
+      *   The computation requiring the function implementation
+      * @param handle
+      *   The stateful implementation to provide
+      * @return
+      *   The computation result with the stateful implementation provided
+      */
     inline def handleState[I[_], O[_], E <: ArrowEffect[I, O], State, A, B, S, S2, S3](
-        inline tag: Tag[E],
+        inline effectTag: Tag[E],
         inline state: State,
         v: A < (E & S)
     )(
         inline handle: [C] => (I[C], State, Safepoint ?=> O[C] => A < (E & S & S2)) => (State, A < (E & S & S2)) < S3,
-        inline done: (State, A) => B < (S & S2 & S3) = (_: State, v: A) => v,
-        inline accept: [C] => I[C] => Boolean = [C] => (v: I[C]) => true
-    )(using inline _frame: Frame, inline flat: Flat[A], safepoint: Safepoint): B < (S & S2 & S3) =
+        inline done: (State, A) => B < (S & S2 & S3) = (_: State, v: A) => v
```

Comment:
Do you need `done` here?

Reply:
Yes, it's used in `Var.run` for example.